test_runner/regress/test_pageserver_secondary.py

```diff
@@ -1096,3 +1096,62 @@ def check_archival_state(state: TimelineArchivalState, tline):
     # Warm up the current secondary.
     ps_attached.http_client().tenant_secondary_download(tenant_id, wait_ms=100)
     wait_until(lambda: all_layers_downloaded(ps_secondary, expected_locally))
+
+
+@run_only_on_default_postgres("PG version is not interesting here")
+def test_io_metrics_match_secondary_timeline_lifecycle(neon_env_builder: NeonEnvBuilder):
+    """
+    Check that IO metrics for secondary timelines are de-registered when the timeline
+    is removed
+    """
```

Comment:
This test (and the PR) doesn't cover the when the tenant is detached because that path doesn't call remove_timeline.

Reply:
Indeed. Fixed in https://github.com/neondatabase/neon/pull/11283/commits/2ce36f0c616fd18a24cdf476286c1cf3520b3b75